packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.js

```diff
@@ -444,11 +472,62 @@ export default class QuanticCategoryFacet extends LightningElement {
     return path.join('/');
   }
 
-  setFocusToFirstFacetValue() {
+  /**
+   * Sets the focus on the target element.
+   */
+  setFocusOnTarget() {
+    this.focusShouldBeInFacet = false;
+    if (!this.focusTarget) {
+      return;
+    }
+    // eslint-disable-next-line default-case
+    switch (this.focusTarget.type) {
+      case 'facetHeader':
+        this.setFocusOnHeader();
+        return;
+      case 'facetValue':
+        if (this.values.length) {
+          this.setFocusOnFirstFacetValue();
+        } else {
+          this.setFocusOnLastNonActiveParent();
+        }
+    }
+  }
+
+  /**
+   * Sets the focus on the first facet value.
+   */
+  setFocusOnFirstFacetValue() {
     const focusTarget = this.template.querySelector(
-      'c-quantic-category-facet-value'
+      '.facet__value > c-quantic-category-facet-value'
+    );
+
+    if (focusTarget) {
+      // @ts-ignore
+      focusTarget.setFocus();
+    }
+  }
+
+  /**
+   * Sets the focus on the last non-active parent.
+   */
+  setFocusOnLastNonActiveParent() {
+    const nonActiveParents = this.template.querySelectorAll(
+      '.facet__non-active-parent > c-quantic-category-facet-value'
     );
+    const lastNonActiveParent = nonActiveParents[nonActiveParents.length - 1];
 
+    if (lastNonActiveParent) {
+      // @ts-ignore
+      lastNonActiveParent.setFocus();
+    }
```

Comment:
```suggestion
    if (nonActiveParents.length > 0) {
      // @ts-ignore
      nonActiveParents[nonActiveParents.length - 1].setFocus();
    }
```

I believe this does pretty much the same thing right?